### PR TITLE
Painless Context Doc: Add field context example

### PR DESCRIPTION
**Pull request by mayya-sharipova**
relates to #34829

### CODE REVIEW

**Review by jdconrad — COMMENTED** [review]
@mayya-sharipova These are great! I'll shoot @debadair a quick ping as I'd like her to look this over from the docs side of things.

File: docs/painless/painless-contexts/painless-field-context.asciidoc [hunk 1]
```diff
@@ -26,3 +26,64 @@ a customized value for each document in the results of a query.
...
+
+
+*Example*
+
+To run this example, first follow the steps in
+<<painless-context-examples, context examples>>.
+
+We will use two scripts to create two additional fields for each search hit.
+These script fields will output computed custom information.
+
```
Comment (debadair) [inline docs/painless/painless-contexts/painless-field-context.asciidoc]: In general, I recommend sticking with the present tense and avoiding "we" except for cases where "we" means "Elastic" (particularly in cases where you're making a recommendation). In this case, I'd probably edit this to read:

You can then use these example scripts to compute custom information and output it to two new fields.

File: docs/painless/painless-contexts/painless-field-context.asciidoc [hunk 2]
```diff
@@ -26,3 +26,64 @@ a customized value for each document in the results of a query.
...
+The first script calculates the day of the week for the field `datetime`:
+
```
Comment (debadair) [inline docs/painless/painless-contexts/painless-field-context.asciidoc]: The callout below is fairly redundant. I'd remove the callout and edit the intro to read:

The first script gets the doc value for the `datetime` field and calls the `getDayOfWeek` function to determine the corresponding day of the week.

File: docs/painless/painless-contexts/painless-field-context.asciidoc [hunk 3]
```diff
@@ -26,3 +26,64 @@ a customized value for each document in the results of a query.
...
+----
+
+<1> Accesses a doc value for the `datetime` field, and uses the function
+    `getDayOfWeek` to calculate the day of the week for this date.
+
+
+
+The second script calculates the number of actors. Actors' names are stored
+as an array in the `actors` field:
+
```
Comment (debadair) [inline docs/painless/painless-contexts/painless-field-context.asciidoc]: I'd say "as a text array" -- then you don't have to reitrate that `actors` is a text field in the callout. 

I'd  also change the :  to a . to be consistent with the punctuation above. (And the second sentence adds info, rather than really leading into the example.)

File: docs/painless/painless-contexts/painless-field-context.asciidoc [hunk 4]
```diff
@@ -26,3 +26,64 @@ a customized value for each document in the results of a query.
...
+[source,Painless]
+----
+params['_source']['actors'].length;
+----
+
+<1> As `actors` field is of type `text`, by default, doc values are
+    not available for this field. We parse `_source` to extract
+    `actors` and calculate their number. Note how `params['_source']['actors']`
+    is a list.
+
```
Comment (debadair) [inline docs/painless/painless-contexts/painless-field-context.asciidoc]: I'd rephrase this slightly:

By default, doc values are not available for text fields. However, you can still calculate the number of actors by extracting `actors` from  `_source`. Note that `params['_source']['actors']` is a list.

**Review by debadair — CHANGES_REQUESTED** [review]
Suggested some minor edits.

**Comment by mayya-sharipova** [comment]
@debadair  Thank you for your review. I have addressed your comments the way you suggested. Can you please let me know if there is anything else that needs to be done.

**Review by debadair — APPROVED** [review]
